pgx/src/spi.rs

```diff
@@ -487,6 +488,52 @@ impl SpiTupleTable {
             }
         }
     }
+
+    /// Returns the number of columns
+    pub fn columns(&self) -> usize {
+        match self.tupdesc {
+            Some(tupdesc) => unsafe { (*tupdesc).natts as usize },
+            None => 0,
+        }
+    }
+
+    /// Returns column type OID
+    ///
+    /// The ordinal position is 1-based
+    pub fn column_type_oid(&self, ordinal: usize) -> Option<PgOid> {
```

Comment:
`usize` should probably be `NonZeroUsize`.

Not very ergonomic, but no surprising behaviors around zero.